fs/sync/sync.go

```diff
@@ -1203,7 +1203,7 @@ func (s *syncCopyMove) setDelayedDirModTimes(ctx context.Context) error {
 			g.Go(func() error {
 				var err error
 				// if item.src is set must copy full metadata
-				if item.src != nil {
+				if item.src != nil && s.setDirMetadata {
```

Comment:
I have now fixed this is a slightly different way in 39f8d039fe07c0215dba8bd192edbadf53c3d091 to make it clearer what is going on rather than using the state of `item.src == nil` to signify metadata.